test/unit/nics/base_test.rb

```diff
@@ -106,10 +106,25 @@ class NicBaseTest < ActiveSupport::TestCase
         nic
       end
 
+      let(:nic2) do
+        host.interfaces.build(:managed => true, :type => 'Nic::Managed')
+      end
+
       test 'it is invalid because of conflicting identifier' do
         refute nic.valid?
         assert nic.errors.has_key?(:identifier)
       end
+
+      test 'it ignores empty identifiers' do
+        nic.mac = '00:11:11:22:22:33'
+        nic2.mac = '00:11:11:22:22:34'
+        nic.identifier = nic2.identifier = ''
+        nic.save
+        nic2.save
```

Comment:
nitpick - nic2.save doesn't need to be called here, just `.valid?` will fail without the patch as there exists a `nic` with identifier `' '`. I'll remove it on merge if jenkins is :green_apple: